src/utils/duration.ts

```diff
@@ -0,0 +1,23 @@
+import assert from "assert";
+
+/**
+ * Formats duration with a format like "2h 25m 12s".
+ * @param start timestamp in milliseconds
+ * @param end timestamp in milliseconds
+ */
+export const formatDuration = (start: number, end: number): string => {
+  const totalSeconds = (end - start) / 1000;
```

Comment:
`Math.floor` on this one?

My point: small optimization to only use integer remainder `%` operator (not the same function used at runtime)